storage/post-storage-logs-to-log-analytics/PostStorageLogs2LogAnalytics.ps1

```diff
@@ -0,0 +1,315 @@
+        }
+
+        if($StartConvert -eq $true -and $c -eq ';')
+        {
+            $ReturnText += "%3B"
+        } else {
+            $ReturnText += $c
+        }
+    }
+
+    return $ReturnText
+}
+
+#
+# If a text doesn't start with ", add "" for json value format
+# If a text contains "%3B", replace it back to ";"
+#
+Function FormalizeJsonValue($Text)
+{
+    $Text1 = ""
+    if($Text.IndexOf("`"") -eq 0) { $Text1=$Text } else {$Text1="`"" + $Text+ "`""}
+
+    if($Text1.IndexOf("%3B") -ge 0) {
+        $ReturnText = $Text1.Replace("%3B", ";")
+    } else {
+        $ReturnText = $Text1
+    }
+    return $ReturnText
+}
+
+Function ConvertLogLineToJson([String] $logLine)
+{
+    #Convert semicolon to %3B in the log line to avoid wrong split with ";"
+    $logLineEncoded = ConvertSemicolonToURLEncoding($logLine)
+
+    $elements = $logLineEncoded.split(';')
+
+    $FormattedElements = New-Object System.Collections.ArrayList
+                
+    foreach($element in $elements)
+    {
+        # Validate if the text starts with ", and add it if not
+        $NewText = FormalizeJsonValue($element)
+
+        # Use "> null" to avoid annoying index print in the console
+        $FormattedElements.Add($NewText) > null
+    }
+
+    $Columns = 
+    (   "version-number",
+        "request-start-time",
+        "operation-type",
+        "request-status",
+        "http-status-code",
+        "end-to-end-latency-in-ms",
+        "server-latency-in-ms",
+        "authentication-type",
+        "requester-account-name",
+        "owner-account-name",
+        "service-type",
+        "request-url",
+        "requested-object-key",
+        "request-id-header",
+        "operation-count",
+        "requester-ip-address",
+        "request-version-header",
+        "request-header-size",
+        "request-packet-size",
+        "response-header-size",
+        "response-packet-size",
+        "request-content-length",
+        "request-md5",
+        "server-md5",
+        "etag-identifier",
+        "last-modified-time",
+        "conditions-used",
+        "user-agent-header",
+        "referrer-header",
+        "client-request-id"
+    )
+
+    # Propose json payload
+    $logJson = "[{";
+    For($i = 0;$i -lt $Columns.Length;$i++)
+    {
+        $logJson += "`"" + $Columns[$i] + "`":" + $FormattedElements[$i]
+        if($i -lt $Columns.Length - 1) {
+            $logJson += ","
+        }
+    }
+    $logJson += "}]";
+
+    return $logJson
+}
+
+$storageAccount = Get-AzureRmStorageAccount -ResourceGroupName $ResourceGroup -Name $StorageAccountName -ErrorAction SilentlyContinue
+if($storageAccount -eq $null)
+{
+    throw "The storage account specified does not exist in this subscription."
+}
+
+$storageContext = $storageAccount.Context
+$containers = New-Object System.Collections.ArrayList
+$container = Get-AzureStorageContainer -Context $storageContext -Name "`$logs" -ErrorAction SilentlyContinue |
```

Comment:
The token ``"`$logs"`` here doesn't appear to be set anywhere or evaluate to something standard, as evidenced by running a few snippets from the script:

```
PS C:\Users\sttramer> $StorageAccountName = "psstoragetest"
PS C:\Users\sttramer> $ResourceGroup = "DefaultRG"
PS C:\Users\sttramer> $storageAccount = Get-AzureRmStorageAccount -ResourceGroupName $ResourceGroup -Name $StorageAccountName
PS C:\Users\sttramer> $storageContext = $storageAccount.Context
PS C:\Users\sttramer> $containers = New-Object System.Collections.ArrayList
PS C:\Users\sttramer> $container = Get-AzureStorageContainer -Context $storageContext -Name "`$logs"
Get-AzureStorageContainer : Can not find the container '$logs'.
At line:1 char:14
+ ... container = Get-AzureStorageContainer -Context $storageContext -Name  ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (:) [Get-AzureStorageContainer], ResourceNotFoundException
    + FullyQualifiedErrorId : ResourceNotFoundException,Microsoft.WindowsAzure.Commands.Storage.Blob.Cmdlet.GetAzureStorageContainerCommand
```

Script should be fixed so that this token is correct.